syntax-sloc/syntax-predicates.rkt

```diff
@@ -0,0 +1,97 @@
+#lang typed/racket/base
+
+(provide module-path*->stx-predicate
+         module-path->stx-predicate
+         is-type-ann?
+         is-contract?)
+
+(require racket/set
+         syntax-sloc/syntax-sloc)
+
+(require/typed syntax/id-set
+  (#:opaque Free-Id-Set immutable-free-id-set?)
+  (immutable-free-id-set (->* [] [(Listof Identifier)] Free-Id-Set))
+  (free-id-set-add (-> Free-Id-Set Identifier Free-Id-Set))
+  (free-id-set->list (-> Free-Id-Set (Listof Identifier)))
+  (free-id-set-member? (-> Free-Id-Set Identifier Boolean))
+  (free-id-set-for-each (-> Free-Id-Set (-> Identifier Void) Void))
+  (free-id-set-union (-> Free-Id-Set Free-Id-Set Free-Id-Set))
+)
+
+(require/typed syntax/modresolve
+  (resolve-module-path (-> Module-Path Path)))
+
+(module+ test
+  (require typed/rackunit))
+
+(: module-path*->stx-predicate (-> (Listof Module-Path) Stx-Pred))
+(define (module-path*->stx-predicate mp*)
+  (define p* (map module-path->stx-predicate mp*))
+  (lambda ([stx : (Syntaxof Any)])
+    (for/or ([p (in-list p*)]) : Boolean
+      (and p (p stx)))))
+
+(: module-path->stx-predicate (-> Module-Path Stx-Pred))
+(define (module-path->stx-predicate mp)
+  (define id* (module-path->id-set mp))
+  (lambda ([stx : (Syntaxof Any)])
+    (and (identifier? stx)
+         (free-id-set-member? id* stx)
```

Comment:
This works, I guess, because `stx` has no scope attached.
(That's also why it's inaccurate.)